src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -225,14 +231,17 @@ public Response addGzip(InputStream gzip)
 			JsonResponseBuilder builder = new JsonResponseBuilder(Response.Status.BAD_REQUEST);
 			return builder.addError(e.getMessage()).build();
 		}
-		return (add(gzipInputStream));
+		return (add(httpHeaders, gzipInputStream));
 	}
 
 	@POST
 	@Produces(MediaType.APPLICATION_JSON + "; charset=UTF-8")
 	@Path("/datapoints")
-	public Response add(InputStream json)
+	public Response add(@Context HttpHeaders httpHeaders, InputStream json)
 	{
+		Scope scope = activateScope("datapoints", httpHeaders);
```

Comment:
``datapoints`` name seems so generic. May be we can pick something more [descriptive](https://github.com/opentracing/specification/blob/master/specification.md#start-a-new-span)